Patch taken from upstream, this can be removed at next update.

commit fc5e905dc90e21b1a381bde42e22c06f45c17e16
Author: Michael Albinus <michael.albinus@gmx.de>
Date:   Tue Apr 22 08:45:34 2025 +0200

    Make `man' more portable
    
    * lisp/man.el (Man-init-defvars): Use [:cntrl:].  Character ranges
    with octal numbers aren't portable enough.  (Bug#77944)

diff --git a/lisp/man.el b/lisp/man.el
index d34d9154052..397162a7ad1 100644
--- a/lisp/man.el
+++ b/lisp/man.el
@@ -650,9 +650,7 @@ This is necessary if one wants to dump man.el with Emacs."
 	     (if Man-sed-script
 		 (concat "-e '" Man-sed-script "'")
 	       "")
-             ;; Use octal numbers.  Otherwise, \032 (Ctrl-Z) would
-             ;; suspend remote connections.
-	     "-e '/^[\\o001-\\o032][\\o001-\\o032]*$/d'"
+	     "-e '/^[[:cntrl:]][[:cntrl:]]*$/d'"
 	     "-e '/\e[789]/s///g'"
 	     "-e '/Reformatting page.  Wait/d'"
 	     "-e '/Reformatting entry.  Wait/d'"
@@ -785,7 +785,7 @@ Different man programs support this feature in different ways.
 The default Debian man program (\"man-db\") has a `--local-file'
 \(or `-l') option for this purpose.  The default Red Hat man
 program has no such option, but interprets any name containing
-a \"/\" as a local filename.  The function returns either `man-db'
+a \"/\" as a local filename.  The function returns either `man-db',
 `man', or nil."
   (if (eq Man-support-local-filenames 'auto-detect)
       (with-connection-local-variables
